geniza/corpus/management/commands/import_data.py

```diff
@@ -41,34 +48,72 @@
     }
 }
 
+# events in document edit history with missing/malformed dates will replace
+# missing portions with values from this date
+DEFAULT_EVENT_DATE = datetime(2020, 1, 1)
+
+# logging config: use levels as integers for verbosity option
+logger = logging.getLogger("import")
+logging.basicConfig()
+LOG_LEVELS = {
+    0: logging.ERROR,
+    1: logging.WARNING,
+    2: logging.INFO,
+    3: logging.DEBUG
+}
+
 
 class Command(BaseCommand):
     'Import existing data from PGP spreadsheets into the database'
 
-    logentry_message = 'Created via data import script'
+    logentry_message = 'Imported via script'
 
     content_types = {}
     collection_lookup = {}
     document_type = {}
     language_lookup = {}
+    user_lookup = {}
     max_documents = None
 
     def add_arguments(self, parser):
         parser.add_argument('-m', '--max_documents', type=int)
 
-    def setup(self):
+    def setup(self, *args, **options):
         if not hasattr(settings, 'DATA_IMPORT_URLS'):
-            raise CommandError('Please configure DATA_IMPORT_URLS in local settings')
-
+            raise CommandError(
+                'Please configure DATA_IMPORT_URLS in local settings')
+
+        # setup logging; default to WARNING level
+        verbosity = options.get("verbosity", 1)
+        logger.setLevel(LOG_LEVELS[verbosity])
+
+        # load fixure containing known historic users (all non-active)
+        call_command("loaddata", "historic_users",
+                     app_label="corpus", verbosity=0)
+        logger.info("loaded 30 historic users")
+
+        # ensure current active users are present, but don't try to create them
+        # in a test environment because it's slow and requires VPN access
+        for username in ["rrichman", "mrustow", "ae5677", "alg4"]:
+            if "PYTEST_CURRENT_TEST" not in os.environ:
+                call_command("createcasuser", username,
+                             staff=True, verbosity=verbosity)
```

Comment:
Why not run as a single command? That works on the command line, does it not work here, or is it slower?

Reply:
didn't even know you could! great suggestion